internal/resources/business_unit/resource.go

```diff
@@ -0,0 +1,934 @@
+			"version": schema.Int64Attribute{
+				Description: "The current version of the Company.",
+				Computed:    true,
+			},
+			"key": schema.StringAttribute{
+				Description: "User-defined unique identifier for the Company.",
+				Required:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Company.",
+				Required:    true,
+			},
+			"status": schema.StringAttribute{
+				Description: "The status of the Company.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(BusinessUnitActive, BusinessUnitInactive),
+				},
+			},
+			"contact_email": schema.StringAttribute{
+				Description: "The email address of the Company.",
+				Optional:    true,
+			},
+			"shipping_address_ids": schema.ListAttribute{
+				Description: "List of the shipping addresses used by the Company.",
+				Optional:    true,
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"billing_address_ids": schema.ListAttribute{
+				Description: "List of the billing addresses used by the Company.",
+				Optional:    true,
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"default_shipping_address_id": schema.StringAttribute{
+				Description: "ID of the default shipping Address.",
+				Optional:    true,
+			},
+			"default_billing_address_id": schema.StringAttribute{
+				Description: "ID of the default billing Address.",
+				Optional:    true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"associate": schema.ListNestedBlock{
+				Description: "Associates that are part of the Business Unit in specific roles",
+				NestedObject: schema.NestedBlockObject{
+					Blocks: map[string]schema.Block{
+						"associate_role_assignments": schema.ListNestedBlock{
+							Description: "Roles assigned to the Associate within a Business Unit.",
+							NestedObject: schema.NestedBlockObject{
+								Attributes: map[string]schema.Attribute{
+									"inheritance": schema.StringAttribute{
+										Description: "Determines whether the AssociateRoleAssignment can be inherited by child Business Units",
+										Required:    true,
+										Validators: []validator.String{
+											stringvalidator.OneOf(
+												AssociateRoleInheritanceEnabled,
+												AssociateRoleInheritanceDisabled,
+											),
+										},
+									},
+								},
+								Blocks: map[string]schema.Block{
+									"associate_role": schema.ListNestedBlock{
+										Description: "Reference to an AssociateRole by its key.",
+										NestedObject: schema.NestedBlockObject{
+											Attributes: map[string]schema.Attribute{
+												"key": schema.StringAttribute{
+													Description: "User-defined unique identifier of the Associate Role",
+													Required:    true,
+												},
+												"type_id": schema.StringAttribute{
```

Comment:
I think this one should be inferred, and we should add id instead, and make both key and id optional